test-crawler/collector/collector.go

```diff
@@ -126,3 +129,31 @@ func ListGoFilesInFolder(root string, ignore []string) (system string, files []s
 
 	return "", nil, nil
 }
+
+// GeneratMethodName generates name adding underscore and usecase index
+// ex: name: ListenProofRequest -> LISTEN_PROOF_REQUEST_001
+func GeneratMethodName(funcName string) string {
+	usecase := "001" // in the future it will be autoincremented depending how many return cases we have.
+	buf := &bytes.Buffer{}
+	for i, rune := range funcName {
+		if unicode.IsUpper(rune) && i > 0 {
+			buf.WriteRune('_')
+		}
+		buf.WriteRune(rune)
+	}
+
+	return fmt.Sprintf("%s_%s", strings.ToUpper(buf.String()), usecase)
+}
+
+// GenerateMethodDescription generates description based on inputParams, returnParams
+func GenerateMethodDescription(f FunctionAnnotation) string {
+	var dsc string
+	if f.Description == "" {
+		dsc = "Function description not set."
+		if (f.InputParams != "" && f.InputParams != "()") && f.ReturnValues != "" {
+			dsc = fmt.Sprintf(`Given a %s, returns %s`, f.InputParams, f.ReturnValues)
+		}
```

Comment:
Why did you remove the case when the function has no input values, but has a return value?

Reply:
I think that's dummy description, better to be default value than just simple 'returns ...' , since we will have 3 cases:
1. GoDoc description if exists
2. Given a input params, returns ...
3. Default value